pkg/apis/pipeline/v1beta1/taskrun_validation_test.go

```diff
@@ -177,6 +284,37 @@ func TestTaskRun_Validate(t *testing.T) {
 			},
 		},
 		wc: config.EnableAlphaAPIFields,
+	}, {
+		name: "propagating partial object params with multiple keys",
+		taskRun: &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "tr"},
+			Spec: v1beta1.TaskRunSpec{
+				Params: []v1beta1.Param{{
+					Name: "task-words",
+					Value: v1beta1.ParamValue{
+						Type:      v1beta1.ParamTypeObject,
+						ObjectVal: map[string]string{"hello": "task run"},
+					},
+				}},
+				TaskSpec: &v1beta1.TaskSpec{
+					Params: []v1beta1.ParamSpec{{
+						Name: "task-words",
+						Type: v1beta1.ParamTypeObject,
+						Default: &v1beta1.ParamValue{
+							Type:      v1beta1.ParamTypeObject,
+							ObjectVal: map[string]string{"hello": "task run def", "hi": "task"},
+						},
+					}},
+					Steps: []v1beta1.Step{{
+						Name:    "echo",
+						Image:   "ubuntu",
+						Command: []string{"echo"},
+						Args:    []string{"$(params.task-words.hello)", "$(params.task-words.hi)"},
+					}},
+				},
+			},
+		},
+		wc: config.EnableAlphaAPIFields,
```

Comment:
Without `properties`, this tricks the [check](https://github.com/tektoncd/pipeline/blob/0fe293118eccdbf4dd478f00ef8d41e0bff06af5/pkg/reconciler/taskrun/validate_resources.go#L86-L88) on the reconciler side to pass because the check compares what are provided on the taskrun level against what are needed in the spec's `properties` section. And the needed is empty so the check is passed.

Although taskrun will not need to provide values for all keys required in `properties` because of https://github.com/tektoncd/pipeline/issues/5270 , the reconciler side will still need to compare things (combined key set provided in taskrun and spec's default) against `properties` section anyway.

Reply:
I now provide properties after our discussion today.

Reply:
Resolve this if you are now satisfied.